### PR TITLE
thetaSketch support

### PR DESCRIPTION
**Pull request by RichRadics**
Add support for thetaSketch aggregators and postAggregators.

### CODE REVIEW

File: pydruid/utils/postaggregator.py [hunk 1]
```diff
@@ -98,3 +98,56 @@ def __init__(self, name):
...
+    def __init__(self, fn, fields, name):
+        self.post_aggregator = {'type': 'thetaSketchSetOp',
+                                'name': name,
+                                'func': fn,
+                                'fields': fields}
+        self.name = name
+
+    def __or__(self, other):
+        return ThetaSketchOp('UNION', self.fields(other),
+                              self.name + 'or' + other.name)
```
Comment (nishantmonu51) [inline pydruid/utils/postaggregator.py]: minor nit: maybe change name to self.name + '_or_' + other.name, in order to make names more readable. 
same comment for and and not

File: pydruid/utils/postaggregator.py [hunk 2]
```diff
@@ -98,3 +98,56 @@ def __init__(self, name):
...
+    def build_post_aggregators(thetasketchops):
+        def rename_thetasketchop(new_name, thetasketchop):
+            thetasketchop['name'] = new_name
+            return thetasketchop
+
+        return [rename_thetasketchop(new_name, thetasketchop.post_aggregator)
+                for (new_name, thetasketchop) in six.iteritems(thetasketchops)]
+
+
+class Theta(ThetaSketchOp):
```
Comment (RichRadics) [inline pydruid/utils/postaggregator.py]: done

File: README.md [hunk 3]
```diff
@@ -151,3 +151,38 @@ def your_asynchronous_method_serving_top10_mentions_for_day(day
     # can be simply ```return top_mentions``` in python 3.x
     raise gen.Return(top_mentions)
 ```
+
+
+#thetaSketches
+Theta sketches post-aggregators are built slightly differently to normal PAs, as they have different operators.
+Note: you must have the ```druid-datasketches``` extension loaded to use these.
```
Comment (RichRadics) [inline README.md]: done

File: README.md [hunk 4]
```diff
@@ -151,3 +151,38 @@ def your_asynchronous_method_serving_top10_mentions_for_day(day
     # can be simply ```return top_mentions``` in python 3.x
     raise gen.Return(top_mentions)
 ```
+
+
+#thetaSketches
+Theta sketches post-aggregators are built slightly differently to normal PAs, as they have different operators.
```
Comment (RichRadics) [inline README.md]: done

File: tests/utils/test_postaggregators.py [hunk 5]
```diff
@@ -0,0 +1,66 @@
+# -*- coding: UTF-8 -*-
```
Comment (RichRadics) [inline tests/utils/test_postaggregators.py]: done

File: pydruid/utils/postaggregator.py (hunk 1 above)
Comment (RichRadics) [inline pydruid/utils/postaggregator.py]: Not sure what you mean here, but I've changed to '_OR_' to make things readable, e.g. output name will be `theta1_OR_theta2`.

**Comment by RichRadics** [comment]
Rebased to latest master, to allow merge

**Comment by RichRadics** [comment]
Rebased again. Are there any blocks to a merge @nishantmonu51 ?

**Comment by joshwalters** [comment]
@RichRadics Will this also support Theta Sketches in arithmetic post-aggs (*, /, etc)? If so, awesome!

**Comment by RichRadics** [comment]
@joshwalters it should work fine - the implementation returns a postAggregator object so should be combinable with the other types perfectly well.

**Comment by joshwalters** [comment]
Just reviewed the code in [Superset](https://github.com/airbnb/superset), this patch should fix the post-agg theta sketch issue we have been seeing. @nishantmonu51 is there anything we can do to help get this merged in?

**Comment by gianm** [comment]
Thanks @RichRadics!

Could you please fill out our CLA here so we can commit your patch: http://druid.io/community/cla.html

**Comment by gianm** [comment]
thanks!

**Pull request merged**